### DISCUSSION TITLE
DuckDB - performance issues on larger tables/files

### DISCUSSION THREAD

**Discussion by reshma-sreekumar** [duckdb/duckdb#10309] [General]
Scenario: Benchmarking performance and spend on 64 million records data that is stored in a duckdb file/table.
CPU and RAM: Intel Core i7 32 GB
Duckdb version: 0.9.0
Results: 
o	Single alter statement – 5.63 s
o	Single update statement – 11.50 sec
o	10 alter statement executed one after the other – 44.33 sec
o	10 update statement executed one after the other – 175 sec (2.9 min)
o	20 alter statement executed one after the other - 109 sec (1.81 min)
o	20 update statement executed one after the other – 405 sec (6.75 min)

Observation: We notice that alter statements are executed quicker but with increased update statements in duckdb table; the performance is deteriorating. 

Could you please suggest how can we improve the performance on larger tables/files

**Comment by Mause** [comment]
Have you tried with 0.9.2?

**Reply by reshma-sreekumar** [reply]
Yes, we did. With 0.9.2 it is taking around 2 mins more than before.  20 update statements got executed in 520.73 (8.67 min) using 0.9.2.
(But we have observed that even while using 0.9.0 the execution time did increase for subsequent query executions).

**Reply by reshma-sreekumar** [reply]
Is this the expected response time from DuckDB for larger datasets (close to 8 mins for 20 update queries on 64 million records)?
If so, then would be great to know if there is going to be a upcoming release which would give a better performance.
If not, could you please suggest if there are any recommended approaches for optimizing the query execution time.

Current approach followed: We are executing the update operations from Python and we are establishing a new DuckDB connection prior to each query execution (in this case 20 update queries are executed in a for loop). We are closing the connection after each query execution inside the loop.

**Comment by reshma-sreekumar** [comment]
Hi, May I know if there are any suggestions on improving the query execution time?
